### ISSUE TITLE
unused var when linting `babashka.process/$`

### ISSUE THREAD

**Issue by bn-darindouglass** [clj-kondo/clj-kondo#1368]
**version**

v2021.08.06

**platform**

MacOS Catalina, native clj-kondo

**editor**

Emacs + flycheck

**problem**

clj-kondo incorrectly flags bindings used in the options map to `$` as unused. This manifests several ways:

- unused bindings
- unused private vars

side note: the `babashka.process` repo still has [deprecated kondo config for the `$` macro](https://github.com/babashka/process/blob/master/.clj-kondo/hooks/dollar.clj)

**repro**

```clojure
;; foo.clj
(ns foo
  (:require [babashka.process :refer [$] :as p]))

(defn example
  []
  (let [env {"SOME_VAR" "wee"}]
    ;; unused binding env
    (-> ($ {:out :string :env env} echo "hi")
        (p/check)
        (:out)
        (println))))
```

**config**

```clojure
{}
```

**expected behavior**

clj-kondo should recognize `env` as being used.
